tests/output_parser_functional_spec.lua

```diff
@@ -0,0 +1,138 @@
+local nio = require("nio")
+local async = require("neotest.async")
+local neotest_kotlin = require("neotest-kotlin")
+
+describe("output_parser functional", function()
+	local example_project_path = vim.fs.joinpath(debug.getinfo(1).source:match("@?(.*/)"), "example_project")
+	local init_script_path = vim.api.nvim_get_runtime_file("test-logging.init.gradle.kts", false)[1]
+	assert.is_not_nil(init_script_path)
+	init_script_path = vim.fs.abspath(init_script_path)
+
+	local tests_path = vim.fs.joinpath(example_project_path, "app", "src", "test", "kotlin", "org", "example")
+
+	local funspec_file = vim.fs.joinpath(tests_path, "KotestFunSpec.kt")
+	local shouldspec_file = vim.fs.joinpath(tests_path, "KotestShouldSpec.kt")
+	local describespec_file = vim.fs.joinpath(tests_path, "KotestDescribeSpec.kt")
+	local stringspec_file = vim.fs.joinpath(tests_path, "KotestStringSpec.kt")
+	local expectspec_file = vim.fs.joinpath(tests_path, "KotestExpectSpec.kt")
+	local freespec_file = vim.fs.joinpath(tests_path, "KotestFreeSpec.kt")
+	local featurespec_file = vim.fs.joinpath(tests_path, "KotestFeatureSpec.kt")
+	local annotationspec_file = vim.fs.joinpath(tests_path, "KotestAnnotationSpec.kt")
+
+	nio.tests.it("functional test", function()
+		---@type string
+		local results_path = async.fn.tempname() .. ".txt"
+
+		local cmd = string.format(
+			"kotest_filter_specs='%s' kotest_filter_tests='%s' %s/gradlew -p %s -I %s test --console=plain | tee -a %s",
+			"org.example.*",
+			"*",
+			example_project_path,
+			example_project_path,
+			init_script_path,
+			results_path
+		)
+
+		local success, _, _ = os.execute(cmd)
+		assert.is_true(success == 0)
+
+		local spec = {
+			context = {
+				results_path = results_path,
+				path = tests_path,
+			},
+		}
+
+		local results = neotest_kotlin.results(spec, {}, {})
+		local ids = vim.tbl_keys(results)
+		assert.equals(31, #ids)
+
+		-- KotestDescribeSpec.kt
+		assert.equals(6, #vim.tbl_filter(function(value)
+			return vim.startswith(value, describespec_file)
+		end, ids))
+
+		assert.equals("passed", results[describespec_file .. "::a namespace::should handle passed assertions"].status)
+		assert.equals("failed", results[describespec_file .. "::a namespace::should handle failed assertions"].status)
+		assert.equals("skipped", results[describespec_file .. "::a namespace::should handle skipped assertions"].status)
+		assert.equals(
+			"passed",
+			results[describespec_file .. "::a namespace::a nested namespace::should handle passed assertions"].status
+		)
+		assert.equals(
+			"failed",
+			results[describespec_file .. "::a namespace::a nested namespace::should handle failed assertions"].status
+		)
+		assert.equals(
+			"skipped",
+			results[describespec_file .. "::a namespace::a nested namespace::should handle skipped assertions"].status
+		)
```

Comment:
Full assertions for all tests in every format the plugin currently supports. Now with real output every time in CI

Reply:
This is a thing of beauty